### ISSUE TITLE
`immediateStop` does not work as expected for `Http2`

### ISSUE THREAD

**Issue by dilanSachi** [ballerina-platform/ballerina-standard-library#3553]
**Description:**
```ballerina
public function main() returns error? {
    http:Listener 'listener = check new(9090);
    http:Service 'service = service object {
        int i = 0;
        resource function get bar () returns string {
            self.i = self.i + 1;
            return "Hello " + self.i.toString();
        }
    };
    check 'listener.attach('service, "hello");
    check 'listener.'start();

    http:Client 'client = check new("localhost:9090");
    string res = check 'client->get("/hello/bar");
    io:println(res);

    check 'listener.immediateStop();
    runtime:sleep(5);
    res = check 'client->get("/hello/bar");
    io:println(res);
}
```
In the above code, the expected output is only `Hello 1`. However, the output is `Hello 1
Hello 2`. Noticed this when adding tests for `immediateStop` in gRPC listener (uses the same immediateStop implementation of `http`).

**Affected Versions:**
2.5.0-20221020-145800-ad2461d

**Comment by TharmiganK** [comment]
Tested the above code with HTTP 1.1 listener and I am getting `error: Something wrong with the connection` for the second request as expected.

**Comment by dilanSachi** [comment]
This issue is not reproducible in the latest pack (http:2.5.0). Hence closing.

**Comment by dilanSachi** [comment]
This issue can be reproduced when we use 2 `http:Client`s for above scenario.

**Comment by TharmiganK** [comment]
The reproducible code :
```balllerina
import ballerina/io;
import ballerina/lang.runtime;
import ballerina/http;

public function main() returns error? {
    http:Listener 'listener = check new(9090);
    http:Service 'service = service object {
        resource function get hello () returns string {
            return "Hello ";
        }
    };
    check 'listener.attach('service);
    check 'listener.'start();

    http:Client client1 = check new("localhost:9090");
    string res = check client1->get("/hello");
    io:println(res);

    check 'listener.immediateStop();
    runtime:sleep(5);

    http:Client client2 = check new("localhost:9090");
    res = check client2->get("/hello");
    io:println(res);
}
```

Run the file using `bal run main.bal`
Output :
```
Hello
Hello
```
Expected :
```
Hello
error: Something wrong with the connection
```

This is working as expected with HTTP/1.1 listener and giving the above output for HTTP/2 listener

**Comment by TharmiganK** [comment]
This fix is need for `2201.2.x` as well

**Comment by TharmiganK** [comment]
Fixed via https://github.com/ballerina-platform/module-ballerina-http/pull/1351
